collect_app/src/main/java/org/odk/collect/android/formentry/audit/IdentityPromptViewModel.java

```diff
@@ -72,4 +75,8 @@ private static boolean userIsValid(String user) {
     public String getFormTitle() {
         return formName;
     }
+
+    public void exit() {
+        FileUtils.purgeMediaPath(instanceFolder);
```

Comment:
I'm wondering if there's more of a shared place we can handle this clean up. It looks like you can't call `FormSaveViewModel#ignoreChanges` in this scenario as the `FormSaveViewModel` won't know about the `FormController` yet. 

Where does this instance folder actually get created?

Reply:
> Where does this instance folder actually get created?

It's here https://github.com/getodk/collect/blob/master/collect_app/src/main/java/org/odk/collect/android/activities/FormFillingActivity.java#L1960

> I'm wondering if there's more of a shared place we can handle this clean up.

I was also thinking about that and I'm aware of the fact that this solution is maybe not perfect but I wasn't able to find a better one. If there is a form that requires identity we don't even save `formController` before passing it [formControllerAvailable](https://github.com/getodk/collect/blob/master/collect_app/src/main/java/org/odk/collect/android/activities/FormFillingActivity.java#L1979) so it's not available in other places to read the dir parth and remove.

Reply:
Maybe we could pull a chunk of the code in `FormSaveViewModel#ignoreChanges` out into a use case that can be called in both places? Thinking about that 

I'm actually wondering whether if the [clean up code in `ignoreChanges`](https://github.com/getodk/collect/blob/eba8151f4125f1e4cc0c40cf4b66c07231997ffb/collect_app/src/main/java/org/odk/collect/android/formentry/saving/FormSaveViewModel.java#L160) should just be called whenever we end form filling? It might be that we could start thinking of `FormSessionRepository` more like one of our data services and move logic like that in there (it could be triggered from `clear` for example). 

Maybe I'm thinking about this incorrectly, but the bug doesn't feel high priority enough to not take these opportunities to restructure while fixing it.

Reply:
FormSaveViewModel as the name suggest is responsible for saving-related operations. using it in this case where we don't even enter the form-filling view sounds a little bit weird to me. Do you really think it would be a better place?